### PR TITLE
Improve things

### PR DESCRIPTION
**Pull request by huacnlee**
- Add `npm run dev` or `bun run dev` command to preview highlight.
- Fix for better highlight for normal text (See preview screenshot).
- Add negative number format for grammar.

## Screenshot

image

### CODE REVIEW

File: grammar.js [hunk 1]
```diff
@@ -25,14 +25,16 @@ module.exports = grammar({
...
-      $.integer,
-    ),
-    integer: $ => /\d+/,
+    _number: $ => seq(
+      optional("-"), 
+      choice(
+        $.hex,
+        $.float,
+        $.integer,
+    )),
```
Comment (arnau) [inline grammar.js]: This expression doesn't need to change. Both `$.float` and `$.integer` already handle the optional `-`.

File: grammar.js [hunk 2]
```diff
@@ -25,14 +25,16 @@ module.exports = grammar({
...
+    integer: $ => /[\-]?\d+/,
```
Comment (arnau) [inline grammar.js]:
```suggestion
    integer: $ => /-?\d+/,
```

File: grammar.js [hunk 3]
```diff
@@ -25,14 +25,16 @@ module.exports = grammar({
...
     hex: $ => /0[xX][\da-fA-F]+/,
-    float: $ => /(0|[1-9]\d*)\.\d+/,
+    float: $ => /[\-]?(0|[1-9]\d*)\.\d+/,
```
Comment (arnau) [inline grammar.js]:
```suggestion
    float: $ => /-?(0|[1-9]\d*)\.\d+/,
```

File: package.json [hunk 4]
```diff
@@ -3,16 +3,13 @@
...
-    "csv"
-  ],
+  "keywords": ["parser", "tree-sitter", "csv"],
   "author": "Arnau Siches",
   "license": "MIT",
   "scripts": {
     "build": "tree-sitter generate && node-gyp build",
-    "test": "tree-sitter test"
+    "test": "tree-sitter test",
+    "dev": "npm run build && tree-sitter highlight --html examples/*.csv > target/out.html && open target/out.html"
```
Comment (arnau) [inline package.json]:
```suggestion
    "dev": "npm run build && tree-sitter highlight examples/*.csv"
```

I'd rather not have to generate html and have to manage the `target/` folder. Highlighting in the shell should suffice.

File: queries/highlights.scm [hunk 5]
```diff
@@ -9,12 +9,10 @@
...
-(string) @string
 
-(escape_sequence) @constant.character.escape
+((string) @string (#match? @string "^\".+?\""))
 
+(escape_sequence) @constant.character.escape
 
 ; Punctuation
 (delimiter) @punctuation.delimiter
-(quote) @punctuation.bracket
```
Comment (arnau) [inline queries/highlights.scm:19]: I want to keep brackets distinct from the string content. Please keep this line.

File: queries/highlights.scm [hunk 6]
```diff
@@ -9,12 +9,10 @@
   (null)
   (na)
 ] @constant.builtin
-(string) @string
 
-(escape_sequence) @constant.character.escape
+((string) @string (#match? @string "^\".+?\""))
```
Comment (arnau) [inline queries/highlights.scm]:
```suggestion
(string) @string
```

All strings should be accounted for, both quoted and unquoted.

**Review by arnau — CHANGES_REQUESTED** [review]
@huacnlee thanks for the contribution, I have a few comments that should be addressed before I merge it though.

Also, we should have tests. I noticed that new versions of tree-sitter have moved the corpus into a new folder structure so I'll address it separately.

Ah, forgot to mention: Please remove the bun lock file, the npm lock file should suffice.

File: queries/highlights.scm (hunk 6 above)
Comment (huacnlee) [reply queries/highlights.scm]: If we do that, the all text will be in string style "green", That means there is no highlight diffenrence.

File: package.json (hunk 4 above)
Comment (huacnlee) [reply package.json]: That better to use, thanks for this tip.

image

File: queries/highlights.scm (hunk 6 above)
Comment (huacnlee) [reply queries/highlights.scm]: image

Comment (arnau) [reply queries/highlights.scm]: correct, both are strings so I don't expect there to be any difference. By preserving `(quote) @punctuation.bracket` you'll have quotes in a different colour so they'll serve as a visual marker (assuming the colour theme does not force the same colour of course).

**Comment by huacnlee** [comment]
New update:

image

**Comment by huacnlee** [comment]
Looks all reviews is resolved, please check again.

**Review by arnau — CHANGES_REQUESTED** [review]
Thanks, one last thing: could you remove the `bun.lockb` file please? the rest looks ready 👍

**Comment by huacnlee** [comment]
Done

**Review by arnau — APPROVED** [review]

**Pull request merged**